composer.json

```
@@ -136,7 +136,8 @@
   "scripts": {
     "githooks": [
       "php -r \"if(file_exists('./.git')&&file_exists('./build/hooks/pre-commit'.(PHP_OS=='WINNT'?'.win':''))){copy('./build/hooks/pre-commit'.(PHP_OS=='WINNT'?'.win':''),'./.git/hooks/pre-commit');} if(file_exists('./.git')&&file_exists('./build/hooks/post-checkout')){copy('./build/hooks/post-checkout','./.git/hooks/post-checkout');}\"",
-      "php -r \"if(file_exists('./.git/hooks/pre-commit')&&(PHP_OS!='WINNT')){chmod('./.git/hooks/pre-commit',0755);} if(file_exists('./.git/hooks/post-checkout')&&(PHP_OS!='WINNT')){chmod('./.git/hooks/post-checkout',0755);}\""
+      "php -r \"if(file_exists('./.git/hooks/pre-commit')&&(PHP_OS!='WINNT')){chmod('./.git/hooks/pre-commit',0755);} if(file_exists('./.git/hooks/post-checkout')&&(PHP_OS!='WINNT')){chmod('./.git/hooks/post-checkout',0755);}\"",
+      "php -r \"if(file_exists('./.git/hooks/pre-commit')&&(PHP_OS!='WINNT')){copy('./build/hooks/pre-merge-commit','./.git/hooks/pre-merge-commit'); chmod('./.git/hooks/pre-merge-commit',0755);}\""
```

```suggestion
      "php -r \"if(file_exists('./.git/hooks/pre-merge-commit')&&(PHP_OS!='WINNT')){copy('./build/hooks/pre-merge-commit','./.git/hooks/pre-merge-commit'); chmod('./.git/hooks/pre-merge-commit',0755);}\""
```

---

@matbcvo the change is correct: look for the pre-commit hook, and if the file is there - copy a new version of pre-merge-commit. This way the merge hook will be applied only if the pre-commit hook exists.

---

@biozshock I understand your approach, but I suggest adding a check to ensure that `build/hooks/pre-merge-commit` exists before copying it over, similar to what the check in line 138 does for `build/hooks/pre-commit` and `build/hooks/post-checkout`.

```suggestion
      "php -r \"if(file_exists('./.git/hooks/pre-commit')&&file_exists('./build/hooks/pre-merge-commit')&&(PHP_OS!='WINNT')){copy('./build/hooks/pre-merge-commit','./.git/hooks/pre-merge-commit'); chmod('./.git/hooks/pre-merge-commit',0755);}\""
```

---

@matbcvo  If the file is not there, then this PR is not merged. Otherwise, if the file is not in build directory, the script must show an error.